src/helpers/client/FormValue.js

```diff
@@ -0,0 +1,109 @@
+/** Handles the core of form values, including:
+ * - Storing both a current and future value
+ * - Tracking when the user manipulates a future value
+ * - Propagating current values to future values if they
+ * haven't been manipulated by the user
+ * - Making this instance serializable
+ * - Making sure only valid values are saved.
+ * 
+ * All values stored in a FormValue instance should be valid
+ * 
+ * There must be a better way to write that.
+ * 
+ * @class
+ * @param {string} name - the name of the client property
+ * @param {any} [val] - the initial value
+ * @param {function} [makeValid] - returns a valid value no
+ * matter what. Will be given the new value, the current value,
+ * the string 'current' or 'future', and the `FormValue` instance.
+ * 
+ * @returns FormValue
+ * @todo Finish description of FormValue return value
+ */
+class FormValue {
+
+  constructor ( name, val = null, makeValid ) {
+
+    var fVal = this;
+    
+    fVal.name = name;
+    fVal._futureWasChanged = false;  // Handled internally
+    fVal.makeValid = makeValid || function ( newVal ) { return newVal; };
+    
+    fVal.setCurrent( val );
+
+  }  // End FormValue.constructor()
+
+
+  setCurrent ( val ) {
+
+    var fVal = this;
+
+    // Transformer needed for parseInt and such
+    fVal.current = fVal.makeValid( val, fVal.current, 'current', fVal );
+
+    if ( !fVal._futureWasChanged ) {
+      fVal.setFuture( val, true );
+    }
+
+    return fVal;  // Return valid value instead?
+
+  }  // End FormValue.setCurrent()
+
+
+  setFuture ( val, propagating ) {
```

Comment:
In what scenario would we want use the propagating flag? Also, what does it semantically mean to set a future value as propagating? Where would it propagate to?

Reply:
This flag is to tell `setFuture()` whether it is being called because of a user change or being called internally by something like `setCurrent()`. It is being *propagated to*.

Reply:
Hmm. Wouldn't it make more sense to flag setCurrent to set the value of `_future` directly? Why explicitly expose the `propogating` flag to the outside world to poke at?

Reply:
If we go with this option, maybe you could put together some demo code showing what you envision.